slurm/src/slurmcc/cost.py

```diff
@@ -0,0 +1,256 @@
+import os
+import sys
+import json
+import csv
+from tabulate import tabulate
+from datetime import datetime
+import logging
+import subprocess
+from collections import namedtuple
+from hpc.autoscale.cost.azurecost import azurecost
+
+log = logging.getLogger('cost')
+
+def run_command(cmd: str, stdout=subprocess.PIPE, stderr=subprocess.PIPE):
+    """
+    run arbitrary command
+    """
+    cmd_list = cmd.split(' ')
+    log.debug(cmd_list)
+    try:
+        output = subprocess.run(cmd_list,stdout=stdout,stderr=stderr, check=True,
+                       encoding='utf-8')
+    except subprocess.CalledProcessError as e:
+        log.error(f"cmd: {e.cmd}, rc: {e.returncode}")
+        log.error(e.stderr)
+        sys.exit(1)
+    except Exception as e:
+        log.error(e)
+        raise
+    log.debug(f"output: {output.stdout}")
+    return output
+
+def get_sacct_fields():
+
+    options = []
+    cmd = "/usr/bin/sacct -e"
+    out = run_command(cmd)
+    for line in out.stdout.split('\n'):
+        for opt in line.split():
+            options.append(opt.lower())
+
+    return options
+
+class Statistics:
+
+    def __init__(self):
+
+        self.jobs = 0
+        self.running_jobs = 0
+        self.processed = 0
+        self.unprocessed = 0
+        self.cost_per_sku = {}
+        self.admincomment_err = 0
+
+    def display(self):
+
+        table = []
+        table.append(['Total Jobs', self.jobs])
+        table.append(['Total Processed Jobs', self.processed])
+        table.append(['Total processed running jobs', self.running_jobs])
+        table.append(['Unprocessed Jobs', self.unprocessed])
+        table.append(['Jobs with admincomment errors', self.admincomment_err])
+        print(tabulate(table, headers=['SUMMARY',''], tablefmt="simple"))
+
+class CostSlurm:
+    def __init__(self, start:str, end: str, cluster: str, fmt: str=None) -> None:
+
+        self.start = start
+        self. end = end
+        self.cluster = cluster
+        self.sacct = "/usr/bin/sacct"
+        self.squeue = "/usr/bin/squeue"
+        self.sacctmgr = "/usr/bin/sacctmgr"
+        cache_root = "/tmp"
+        self.stats = Statistics()
+        self.cache = f"{cache_root}/slurm"
+        try:
+            os.makedirs(self.cache, 0o777, exist_ok=True)
+        except OSError as e:
+            log.error("Unable to create cache directory {self.cache}")
```

Comment:
You are missing the leading f here on the format string.

Reply:
fixed